### PR TITLE
Add FAQ entry for how to serve in-memory generated files

### LINKED ISSUE #734: Document how to annotate a generated file response
For a read-only `ViewSet` like the following:

```py
from django.http import HttpResponse

class FileViewSet(RetrieveModelMixin, GenericViewSet):
    ...

    def retrieve(self, request, *args: Any, **kwargs: str) -> HttpResponse:
        ...
        export_data = b"..."
        return HttpResponse(
            export_data,
            headers={
                "Content-Type": "text/plain",
                "Content-Disposition": f"attachment; filename=out.txt",
            },
        )
```

I am using an [`HttpResponse`](https://docs.djangoproject.com/en/4.0/ref/request-response/#httpresponse-objects) to support in-memory generated files to serve to clients.

I am following their instructions to [treat the response as a file](https://docs.djangoproject.com/en/4.0/ref/request-response/#telling-the-browser-to-treat-the-response-as-a-file-attachment).

What is the best way to annotate that this is returning a binary file into a `drf-spectacular` schema? I didn't see any mention of this in the docs. I would be happy to contribute the docs if we are able to find a good solution.

### PR DESCRIPTION
**Pull request by johnthagen**
Closes #734

### CODE REVIEW

**Comment by johnthagen** [comment]
Please squash this PR when merging through the GitHub UI.

File: docs/faq.rst [hunk 1]
```diff
@@ -325,3 +325,39 @@ attribute ``swagger_fake_view`` and simply return an empty queryset of the corre
...
+How to serve in-memory generated files
+--------------------------------------
+
+DRF provides a convenient ``FileField`` for storing files persistently within a ``Model``.
+``drf-spectacular`` handles these correctly by default. But to serve binary files that are
+*generated in memory*, follow the following recipe. This example uses the method
+`recommended by Django <https://docs.djangoproject.com/en/4.0/ref/request-response/#telling-the-browser-to-treat-the-response-as-a-file-attachment>`_
+for treating a ``Response`` as a file and sets up an appropriate ``Renderer`` that will handle the
+client ``Accept`` header for this response content type.
+
```
Comment (tfranzel) [inline docs/faq.rst:338]: would have liked something like "`bytes` states that the response is a binary blob without further details on it's structure"

**Review by tfranzel — COMMENTED** [review]
2 minor comments

**Comment by tfranzel** [comment]
excellent! thx

**Pull request merged**